lib/turbolinks/x_domain_blocker.rb

```diff
@@ -4,18 +4,18 @@ module Turbolinks
   # - The request is being redirected to a different domain
   module XDomainBlocker
     private
-      def same_origin?(a, b)
```

Comment:
@kaspth Let me know if you'd like me to rename to something else, but to me it feels weird that `same_origin?` would return `true` when one of the URIs is invalid (which is the behavior we want here). Or else we put the `rescue` in `abort_xdomain_redirect`.